pyanaconda/ui/gui/spokes/subscription.py

```diff
@@ -390,28 +390,35 @@ def on_http_proxy_checkbox_toggled(self, checkbox):
             # the subscription request structure, but keep it in the entries in case
             # the user tries to show them again before next spoke entry clears them
             self._subscription_request.server_proxy_hostname = ""
-            self._subscription_request.server_proxy_posr = -1
+            self._subscription_request.server_proxy_port = -1
             self._subscription_request.server_proxy_user = ""
-            self._subscription_request.server_proxy_password = ""
+            self._subscription_request.server_proxy_password.set_secret(None)
 
     def on_http_proxy_location_entry_changed(self, editable):
-        hostname = ""
-        port = -1  # not set == -1
-        entered_text = editable.get_text()
-        if entered_text:
-            proxy_obj = ProxyString(url=entered_text)
+        # Incorrect hostnames, including empty strings, will
+        # throw an exception we need to catch and switch
+        # to defaults. This can happen often as the user
+        # types the hostname to the field.
+        try:
+            proxy_obj = ProxyString(url=editable.get_text())
             hostname = proxy_obj.host
             if proxy_obj.port:
                 # the DBus API expects an integer
                 port = int(proxy_obj.port)
+        except ProxyStringError:
+            hostname = ""
+            port = -1  # not set == -1
+        # set the resulting values to the DBus structure
         self.subscription_request.server_proxy_hostname = hostname
         self.subscription_request.server_proxy_port = port
```

Comment:
What happens if no exception is raised, but the condition is not fulfilled?

Reply:
Hmm, that's a good point - while technically this is correct as ProxyString has a default port (3128), thes part of the code still looks confusing with the `port` variable. I'll try to do something about that.